### PR TITLE
New package: hurl-1.0.0

### PR DESCRIPTION
**Pull request by juantascon**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/hurl/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+revision=1
+build_style=cargo
+hostmakedepends="pkg-config"
+makedepends="libxml2-devel libressl-devel"
+short_desc="Hurl, run and test HTTP requests. "
+maintainer="Juan Tascon <juan@horlux.org>"
+license="Apache-2.0"
+homepage="https://github.com/Orange-OpenSource/hurl"
+changelog="https://github.com/Orange-OpenSource/hurl/raw/master/CHANGELOG.md"
+distfiles="https://github.com/Orange-OpenSource/hurl/archive/${version}.tar.gz>${pkgname}-${version}.tar.gz"
```
Comment (abenson) [inline srcpkgs/hurl/template:13]: You don't need to do the redirect here

File: srcpkgs/hurl/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+checksum=3c1bbd3650a91d8434f9a57b0123246c3260e955a966feb2c63dedba94463ac1
+
+do_build() {
+	./ci/man.sh
```
Comment (abenson) [inline srcpkgs/hurl/template:17]: This makes me think its would only generate the man pages and not actually build the binaries?

File: srcpkgs/hurl/template [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'hurl'
+pkgname=hurl
+version=1.0.0
+revision=1
+build_style=cargo
+hostmakedepends="pkg-config"
```
Comment (abenson) [inline srcpkgs/hurl/template:6]:
```
=> hurl-1.0.0_1: running do_build ...
/usr/bin/env: ‘python3’: No such file or directory
```

Probably missing `python3` here.

**Comment by ericonr** [comment]
```
error[E0308]: mismatched types
  --> packages/hurl/src/runner/xpath.rs:88:45
   |
88 |             let x = unsafe { CStr::from_ptr(c_s2) };
   |                                             ^^^^ expected `u8`, found `i8`
   |
   = note: expected raw pointer `*const u8`
              found raw pointer `*const i8`
```

They don't test builds for ARM, this should have been detected a long while ago. Please report to upstream. chars have different default signedness on different platforms, so they should be using `c_char` from `std::os` or `libc`, not `u8` or `i8`.

**Pull request closed without merging**